### PR TITLE
Fix mentions regex

### PR DESCRIPTION
**Pull request by mario**
If this works, it would fix issues like https://github.com/nextcloud/spreed/issues/956.
Signed-off-by: Mario Danic <mario@lovelyhq.com>

### CODE REVIEW

**Comment by mario** [comment]
Seems to fail tests as I didn't take some possible mentions into consideration. Will look into it and try to remove support for things like url/@mention as this is not a proper mention.

**Comment by mario** [comment]
Seems to be good now @blizzz @rullzer and whoever else!

**Comment by skjnldsv** [comment]
Tests would be really appreciated ;)

**Comment by mario** [comment]
@skjnldsv changed regex passes existing Regex tests. If you'd like new ones (which is understandable), I'd appreciate help in the area!

**Comment by skjnldsv** [comment]
@mario could you add the new covered mention syntaxes as well as invalid ones here then?
https://github.com/nextcloud/server/blob/47769d2640be7f49d9db154a161e6c9d84928ee3/tests/lib/Comments/CommentTest.php#L130

**Comment by mario** [comment]
@skjnldsv was hoping to talk you into it xD Let me see what I can do.

**Comment by skjnldsv** [comment]
Nice! :)
Is that it, no more mention formats?

**Comment by mario** [comment]
@skjnldsv added tests. Could you check why Drone fails as it doesn't look like it's failing at my tests? o.O

**Comment by mario** [comment]
@skjnldsv there might be, I don't know xD

**Comment by skjnldsv** [comment]
> Could you check why Drone fails as it doesn't look like it's failing at my tests? o.O
It was a timeout, I restarted the tests :)

**Review by skjnldsv — APPROVED** [review]

**Review by blizzz — APPROVED** [review]
apart from the double quotes

File: tests/lib/Comments/CommentTest.php [hunk 1]
```diff
@@ -144,10 +144,14 @@ public function mentionsProvider() {
 			[
 				'@foobar and @barfoo you should know, @foo@bar.com is valid' .
 					' and so is @bar@foo.org@foobar.io I hope that clarifies everything.' .
-					' cc @23452-4333-54353-2342 @yolo!',
+					' cc @23452-4333-54353-2342 @yolo!' .
+					' however the most important thing to know is that www.croissant.com/@oil is not valid' .
```
Comment (blizzz) [inline tests/lib/Comments/CommentTest.php]: shouldn't this be pancake.com though? 🤡

Comment (mario) [reply tests/lib/Comments/CommentTest.php]: Ideally yes, but I made a honorable mention of @skjnldsv. Though I'm not really sure he can do any of those, he just likes bragging!

**Pull request merged**